### PR TITLE
SC-660: Additional tests

### PR DESCRIPTION
**Pull request by finnoio**

### CODE REVIEW

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PaymentValidationSuite.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+         |{-# SCRIPT_TYPE ASSET #-}
+         |
+         |getStringValue(addressFromString("${dApp.toAddress.toString}").value(), "$wrKey") == "$wrValue"
+         """.stripMargin,
+      isAssetScript = true,
+      ScriptEstimatorV3
+    ).explicitGet()._1.bytes().base64
+    val smartAssetId = sender.issue(caller, script = Some(scr), fee = issueFee + smartFee, waitForTx = true).id
+
+    assertApiError(
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PaymentValidationSuite.scala:49]: move from the prerequisite or rename the test

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala [hunk 2]
```diff
@@ -9,34 +9,64 @@ import com.wavesplatform.lang.v1.compiler.Terms
...
-         |    case r: ReissueTransaction => r.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case b: BurnTransaction => b.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |
+         |    case t: TransferTransaction => t.senderPublicKey.toBase58String() == "${firstDApp.publicKey.toString}"
+         |     && t.assetId.value().toBase58String() == "$firstAssetId"
+         |     && toBase64String(t.attachment) == ""
+         |     && t.bodyBytes.size() == 0
+         |     && t.fee == 0
+         |     && t.id.size() != 0
+         |     && toBase58String(addressFromRecipient(t.recipient).bytes) == "${secondDApp.toAddress.toString}"
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala:41]: What if recipient is alias?

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala [hunk 3]
```diff
@@ -9,34 +9,64 @@ import com.wavesplatform.lang.v1.compiler.Terms
...
          |  match tx {
-         |    case t: TransferTransaction => t.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case r: ReissueTransaction => r.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case b: BurnTransaction => b.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |
+         |    case t: TransferTransaction => t.senderPublicKey.toBase58String() == "${firstDApp.publicKey.toString}"
+         |     && t.assetId.value().toBase58String() == "$firstAssetId"
+         |     && toBase64String(t.attachment) == ""
+         |     && t.bodyBytes.size() == 0
+         |     && t.fee == 0
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala:39]: check feeAssetId

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala [hunk 4]
```diff
@@ -9,34 +9,64 @@ import com.wavesplatform.lang.v1.compiler.Terms
...
          |{-# CONTENT_TYPE EXPRESSION #-}
          |{-# SCRIPT_TYPE ASSET #-}
          |
          |  match tx {
-         |    case t: TransferTransaction => t.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case r: ReissueTransaction => r.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |    case b: BurnTransaction => b.senderPublicKey == base58'${firstDApp.publicKey.toString}'
-         |
+         |    case t: TransferTransaction => t.senderPublicKey.toBase58String() == "${firstDApp.publicKey.toString}"
+         |     && t.assetId.value().toBase58String() == "$firstAssetId"
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala:36]: check Unit for Waves

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala [hunk 5]
```diff
@@ -9,34 +9,64 @@ import com.wavesplatform.lang.v1.compiler.Terms
...
+         |     && toBase64String(t.attachment) == ""
+         |     && t.bodyBytes.size() == 0
+         |     && t.fee == 0
+         |     && t.id.size() != 0
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala:40]: can we calculate it and check exact value?

**Review by msmolyakov — CHANGES_REQUESTED** [review]

**Review by msmolyakov — APPROVED** [review]

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PaymentValidationSuite.scala (hunk 1 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PaymentValidationSuite.scala:49]: done

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala (hunk 2 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala]: done

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala (hunk 3 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala]: done

File: node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala (hunk 5 above)
Comment (finnoio) [reply node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/PseudoTransactionSuite.scala]: done

**Review by xrtm000 — APPROVED** [review]

**Pull request merged**